src/main/java/org/tugraz/sysds/runtime/matrix/data/FrameBlock.java

```diff
@@ -1600,4 +1601,18 @@ public void setMvValue(String mvVal) {
 			_mvValue = mvVal;
 		}
 	}
-}
+
+	public FrameBlock getSchemaTypeOf()
+	{
+		ValueType[] schemaStrings = new ValueType[this.getNumColumns()];
+		for(int i=0; i<schemaStrings.length; i++)
+			schemaStrings[i] = ValueType.STRING;
+
+		String[][] st = new String[1][this.getNumColumns()];
+		ValueType[] schema = this.getSchema();
+
+		for(int i=0; i<schema.length; i++) {st[0][i] = schema[i].toString(); }
+		FrameBlock retBlock = new FrameBlock(schemaStrings, this.getColumnNames(), st);
```

Comment:
directly return the frameblock instead of assigning it to a variable first.

Reply:
Modified as per suggestions. Thank you.